addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -121,6 +121,10 @@ def _create_invoice(self, order, so_line, amount):
                     subtype_id=self.env.ref('mail.mt_note').id)
         return invoice
 
+    @api.model
```

Comment:
wrong design...

create_invoices should return a list of record, and there should be other functions that do:
* returns an action to view the invoices (as currently)
```python
         if self._context.get('open_invoices', False):
              return sale_orders.action_view_invoice()
```
* returns an act_window_close (as currently)
```python
          return {'type': 'ir.actions.act_window_close'}
```
* validate the invoice and do the stuff you try to do
```python
invoices = create_invoices()
invoices.action_invoice_open()  # this function only
move_lines = self.env['account.bank.statement.line'].get_move_lines_for_reconciliation(additional_domain=[('invoice_id', 'in', invoice_ids)])
 return move_lines.prepare_move_lines_for_reconciliation_widget(target_currency=sale_orders[0].currency_id)
```

this way, you can put self-explicit docstrings to the method to explain why you need them and you can remove the fancy crapy context usage there